services/cd-service/pkg/repository/repository.go

```diff
@@ -173,6 +174,90 @@ func openOrCreate(path string, storageBackend StorageBackend) (*git.Repository,
 	return repo2, err
 }
 
+func GetTags(cfg RepositoryConfig, repoName string, ctx context.Context) (tags, commits []string, err error) {
```

Comment:
Please return 1 array with tag and commit:

```suggestion
func GetTags(cfg RepositoryConfig, repoName string, ctx context.Context) (tags []TagData, err error) {
```